components/hab/src/cli.rs

```diff
@@ -1076,6 +1076,12 @@ pub fn sub_sup_run(feature_flags: FeatureFlag) -> App<'static, 'static> {
                                                             Implies NO_COLOR")
                             (@arg HEALTH_CHECK_INTERVAL: --("health-check-interval") -i +takes_value {valid_health_check_interval}
                              "The interval (seconds) on which to run health checks [default: 30]")
+                            (@arg SYS_IP_ADDRESS: --("sys-ip-address") +takes_value
+                             "The fallback IPv4 address if we can not dynamically determine an \
```

Comment:
Yeah, I am not sure what to do here. The [`Sys`](https://github.com/habitat-sh/habitat/blob/332ef8e89ae2abc2af1ec1494d9d92fedb4ce95c/components/sup/src/manager/sys.rs#L18) struct does use `IpAddr`, but from other conversations, I do not believe we support IPv6 (not sure exactly what all the issues are). Also, from discussions around this issue, we only want to look for IPv4 addresses from a network interface.

I see three paths forward:

* Change the types to actually reflect that we only support IPv4. Not sure how much of an impact this would be? Kinda of a step in the wrong direction removing parts of the code that can support either?
* Investigate the amount of work needed for IPv6 support. #5098
* Leave it as is. The help message acts as a warning to the user.

Reply:
Thinking about it more, I think we should make the IP address specified by the user be v4 and leave the existing code as it is.

Reply:
Addressed in [be3aa8](https://github.com/habitat-sh/habitat/pull/6960/commits/be3aa875b5d92f8bfbc6d34142579b2ba2e55d0e).